sdk/core/Azure.Core.TestFramework/src/RecordedTestAttribute.cs

```diff
@@ -102,32 +111,44 @@ public override TestResult Execute(TestExecutionContext context)
                     }
                     else
                     {
-                        // Check if there are any service errors we should ignore.
-                        List<IgnoreServiceErrorAttribute> attributes = Test.GetCustomAttributes<IgnoreServiceErrorAttribute>(true).ToList();
+                        CheckForIgnoredServiceErrors(context);
+                    }
+                }
+                else
+                {
+                    CheckForIgnoredServiceErrors(context);
+                }
```

Comment:
Could this be simplified to a single unconditional `CheckForIgnoredServiceErrors(context);` at the end?

Reply:
today is not my day

Reply:
Wait actually no, because we don't want to apply this if we fall into one of the other error blocks.

Reply:
Would it make sense to return immediately from those other error blocks?